File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points,(a,b)->{
            return Integer.compare(a[0],b[0]);
        });
        List<Pair<Integer,Integer>> ls = new ArrayList<>();
        int s = points[0][0];
        int e = points[0][1];
        for(int i=1;i<points.length;i++){
            if(points[i][0]<=e){
                s = Math.max(s,points[i][0]);
                e = Math.min(e,points[i][1]);
            }
            else{
                Pair<Integer,Integer> p = new Pair<>(s,e);
                ls.add(p);
                s = points[i][0];
                e = points[i][1];
            }
        }
        ls.add(new Pair<Integer,Integer>(s,e));
        return ls.size();
    }
}

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
from functools import cmp_to_key
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key = cmp_to_key(lambda a,b: a[0]-b[0] if a[0]!=b[0] else a[1]-b[1]))

        ans = 1
        right = points[0][1]
        # print(points)

        for i in range(1,len(points)):
            if points[i][0] <= right:
                right = min(points[i][1], right)
            else:
                ans+=1
                right = points[i][1]

        return ans
        